### DISCUSSION TITLE
621

### DISCUSSION THREAD

**Discussion by giscus** [jojoldu/blog-comments#939] [Comments]
# 621

TypeScript의 Enum은 딱 열거형으로서만 사용할 수 있습니다. 다른 언어에서 Enum을 Static 객체로 사용해본 경험이 있는 분들이라면 이 지점이 굉장히 답답하다는 것을 느낄 수 있는데요. Enum을 객체로 활용하면 어떤 큰 장점을 얻게되는지는 배민 기술 블로그 를 참고해보시면 좋습니다. 저와 똑같이 답답함을 느끼신 분이 계시는지, 이미 TypeScript도 Java의 Enum과 같이 Static 객체로 Enum을 다룰 수 있도록 ts-jenum 을 만들어주셨습니다. 이번 시간에는 ts-jenum 을 이용해 응집력 있는 Enum 활용법을 소개드리겠습니다. 1. 설치 ts-jenum 은 Java의 java.lang.Enum 과 같은 사용성을 얻기 위해 제공하는 라이브러리입니다. 별도의 데코레이..

https://jojoldu.tistory.com/621

**Comment by 1wooseok** [comment]
글 너무 잘봤습니다!

혹시 마지막 문장 **라이브러리 특성상 프론트엔드에서 활용하기에는 어려움이 있겠지만** 에서 
생각하시기에 어떤 어려움이 있는지 여쭤봐도 괜찮을까요?!

**Reply by Ssuamje** [reply]
이전에 작성하신 글이어서 넘어가셨을 것 같기도하고, 이미 찾아보셨을 것 같긴하지만...
ts의 enum 사용과 관련해서 프론트에서 tree-shaking에 애로사항이 있는 것 같습니다..! 관련 내용의 글이 꽤 많은 것 같아요.
